enable dumping the 1st plane of ghost cells with DiagFramePlane

baperry2 opened the pull request:
Currently if the requested plane location for the DiagFramePlane Diagnostic is not available, we just silently do nothing. After this PR, we will raise an error, unless the user has specified the `dump_ghost_if_OOB` option, in which case the first ghost cell in the relevant face will be dumped (all levels get their respective first ghost cell). This is intended as debugging capability, for example giving users direct access to visualize what bcnormal is doing. 

In PeleLMeX, the first ghost cell on domain boundaries corresponds to the BC at the boundary **face**.

In PeleC, the first ghost cell on domain boundaries corresponds to the state 1/2 cell over from boundary face, which gets set such that applying standard operators will lead to the desired values on the face.

baperry2 committed:
enable dumping the 1st plane of ghost cells with DiagFramePlane

baperry2 committed:
Merge branch 'development' into diagplane-ghost

malihass reviewed (changes requested):
Thanks for that, it works great in a turbulent channel flow case!
One change that would be good
Can we dump the error message or the info message only once (not MPI process times)?

Also (not for this PR), but it the LMeX doc, it would be nice to have the dump_ghost_if_OOB option.

baperry2 committed:
add diagnostics documentation to PP, moving from LMeX

baperry2 committed:
doc update refinement

baperry2 committed:
use amrex::Print instead of Warning so message only gets written once

baperry2 committed:
Merge branch 'diagplane-ghost' of github.com:baperry2/PelePhysics into diagplane-ghost

baperry2 commented:
Modified so the warning only gets printed once. Leaving the error as is, because we want to ensure an error is always raised if needed.

Moved Docs here so we don't have to maintain separate Docs for this capability in LMeX and C, and added something for the new flag.

malihass reviewed (approved):
Super! Thanks for the change
Yesterday we noticed that the plane at Step 0 was not dumped, so if you want to add this that would be great, otherwise it is already great the way it is!

baperry2 commented:
The change to dump the plane at step 0 will need to go in PeleC/LMeX I think
